apps/src/javalab/JavabuilderConnection.js

```diff
@@ -399,18 +437,7 @@ export default class JavabuilderConnection {
     } else if (this.sawValidationTests) {
       this.onValidationFailed();
     }
-    this.sawValidationTests = false;
```

Comment:
Are these updates still relevant?

Reply:
yep, since I added new flags we need to reset on each run, I made a new method and called it before we start a run rather than here. This code won't always be executed on every run, so it made sense to move it.